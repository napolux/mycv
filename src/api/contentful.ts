import * as Contentful from 'contentful';

export interface IContentfulApiClient {
  client: Contentful.ContentfulClientApi;
  getCv(): Promise<any>;
  mapResponseToPositions(positionsFromResponse: any): Array<ICvPositions>
}

export interface ICvFields {
  name: string;
  surname: string;
  dob: Date;
  phone: string;
  email: string;
  positions: ICvPositions[];
}

export interface ICvPositions {
  company: string;
  role: string;
  description: string;
  start: Date;
  end: Date;
}

export class ContentfulAPIClient implements IContentfulApiClient {

  client: Contentful.ContentfulClientApi;

  constructor(space: string, accessToken: string) {
    this.client = Contentful.createClient({
      space: space,
      accessToken: accessToken
    })
  }

  // We'll get the CV from Contentful
  getCv = async () => {
    try {
      const cv = await this.client.getEntry('2DMIcTZmJvEUe2FFpEQZaT');
      return cv;  
    } catch (error) {
      return {};
    }
  }

  // Utility to map response job positions to output job positions
  mapResponseToPositions(positionsFromResponse: any): Array<ICvPositions> {
    
    let positions: Array<ICvPositions> = [];
    
    positionsFromResponse.forEach((p: any) => {
      positions.push(
        {
          start: p.fields.start,
          end: p.fields.end,
          company: p.fields.company,
          role: p.fields.role,
          description: p.fields.description
        }
      )
    });

    return positions;
  } 
}