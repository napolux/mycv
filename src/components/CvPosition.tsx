import React from 'react';
import { ICvPositions } from '../api/contentful';
import { Divider, Label, Icon, Header, Container } from 'semantic-ui-react'


interface ICvPositionProps {
  cvPositionValue: ICvPositions[];
}

export default class CvPosition extends React.Component<ICvPositionProps> {

  render() {

    // for readability
    let p = this.props.cvPositionValue;

    // job positions to be rendered
    let positions = [];

    for (let i = 0; i < p.length; i++) {
      positions.push(
        <section key={'cvposition_' + i}>
          <div>
            <Header as='h3'>{p[i].company}</Header>
            <Header as='h4'>{p[i].role}</Header>
            <Label>
              <Icon name='calendar' />From
              <Label.Detail>{new Date(p[i].start).toLocaleDateString("it-IT")}</Label.Detail>
            </Label>         
            <Label>
              <Icon name='calendar' />To
              <Label.Detail>{new Date(p[i].end).toLocaleDateString("it-IT")}</Label.Detail>
            </Label>    
            <Container>{p[i].description}</Container>
          </div>
          <Divider />
        </section>
      );
    }


    return (
      <>{positions}</>
    );
  }
}