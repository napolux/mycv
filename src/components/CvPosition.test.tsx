import React from 'react';
import CvPosition from './CvPosition';
import renderer from 'react-test-renderer';
import { ICvPositions } from '../api/contentful';

it('renders correctly', () => {

  const position: ICvPositions[] = [{
    company: 'mock company',
    role: 'mock role',
    description: 'string',
    start: new Date(2018, 11, 12),
    end: new Date(2018, 11, 12),
  }]

  const tree = renderer
    .create(<CvPosition cvPositionValue={position} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});