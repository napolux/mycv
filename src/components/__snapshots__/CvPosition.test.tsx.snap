// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui container"
>
  <div>
    <h3
      className="ui header"
    >
      mock company
    </h3>
    <h4
      className="ui header"
    >
      mock role
    </h4>
    <div
      className="ui label"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="calendar icon"
        onClick={[Function]}
      />
      From
      <div
        className="detail"
      >
        12/24/2018
      </div>
    </div>
    <div
      className="ui label"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="calendar icon"
        onClick={[Function]}
      />
      To
      <div
        className="detail"
      >
        12/24/2018
      </div>
    </div>
    <div
      className="ui container"
    >
      string
    </div>
  </div>
  <div
    className="ui divider"
  />
</div>
`;
